src/TypeInference/SolveConstraints.hs

```diff
@@ -113,7 +133,55 @@ solve (cs:css) = do
       _ -> do
         subCss <- subConstraints cs
         solve (subCss ++ css))
+------------------------------------------------------------------------------
+-- Kind Inference
+------------------------------------------------------------------------------
+unifyKinds :: MonoKind -> MonoKind -> SolverM()
+unifyKinds (CBox cc1) (CBox cc2) = 
+  if cc1 == cc2
+    then return ()
+    else throwSolverError defaultLoc ["Cannot unify incompatible kinds: " <> ppPrint cc1 <> " and " <> ppPrint cc2]
+unifyKinds (KindVar kv1) (KindVar kv2) = do
+  sets <- getKVars
+  let (mmk1, kvs1) = lookupKVar kv1 sets
+  let (mmk2, kvs2) = lookupKVar kv2 sets
+  case (mmk1,mmk2) of 
+    (_, Nothing) -> putKVars $ M.insert mmk1 (S.insert kv2 kvs1) (M.insert mmk2 (S.delete kv2 kvs2) sets)
+    (Nothing, _) -> putKVars $ M.insert mmk2 (S.insert kv1 kvs2) (M.insert mmk1 (S.delete kv1 kvs1) sets)
+    (Just mk1, Just mk2) | mk1 == mk2 -> putKVars sets 
+                         | otherwise -> throwSolverError defaultLoc ["Cannot unify incompatiple kinds: " <> ppPrint mk1 <> " and " <> ppPrint mk2]

```

Comment:
I think there was a copy-paste bug with one of the variable names.
Could you please check if this fix is correct?

Reply:
Looks correct to me